linux/src/platform_linux.c

```diff
@@ -1,13 +1,53 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+#include "c_logging/xlogging.h"
+
 #include "c_pal/platform.h"
+#include "c_pal/completion_port_linux.h"
+
+COMPLETION_PORT_HANDLE g_completion_port = NULL;
 
 int platform_init(void)
 {
-    return 0;
+    int result;
+    if (g_completion_port == NULL)
+    {
+        // Codes_SRS_PLATFORM_LINUX_11_001: [ If the completion_port object is NULL, platform_init shall call completion_port_create. ]
+        g_completion_port = completion_port_create();
+        if (g_completion_port == NULL)
+        {
+            // Codes_SRS_PLATFORM_LINUX_11_003: [ otherwise, platform_init shall return a non-zero value. ]
+            LogError("Failure calling completion_port_create");
+            result = MU_FAILURE;
+        }
+        else
+        {
+            // Codes_SRS_PLATFORM_LINUX_11_002: [ If completion_port_create returns a valid completion port object, platform_init shall return zero. ]
+            result = 0;
+        }
+    }
+    else
+    {
+        // Codes_SRS_PLATFORM_LINUX_11_007: [ If the completion port object is non-NULL, platform_init shall return zero. ]
+        result = 0;
+    }
+    return result;
 }
 
 void platform_deinit(void)
 {
+    // Codes_SRS_PLATFORM_LINUX_11_008: [ If the completion port object is non-NULL, platform_deinit shall do nothing. ]
+    if (g_completion_port != NULL)
+    {
+        // Codes_SRS_PLATFORM_LINUX_11_004: [ If the completion port object is non-NULL, platform_deinit shall decrement the reference by calling completion_port_dec_ref. ]
+        completion_port_dec_ref(g_completion_port);
+        g_completion_port = NULL;
```

Comment:
>g_completion_port = NULL; [](http://example.com/codeflow?start=8&length=25)

wrong always.

Reply:
This has to happen because if you call platform_init g_completion_port is a valid address, then you call platform_deinit which frees the memory (without setting it to NULL).  If you call platform_init again it will not allocate the memory again because g_completion_port will still be a stale address.  So this is why it has to be set to NULL.